server/resource_server/emails/verification/verifier.go

```diff
@@ -0,0 +1,71 @@
+package verification
+
+import (
+	"fmt"
+	"globe-and-citizen/layer8/server/resource_server/constants"
+	"globe-and-citizen/layer8/server/resource_server/emails/sender"
+	"globe-and-citizen/layer8/server/resource_server/emails/verification/code"
+	"globe-and-citizen/layer8/server/resource_server/models"
+	"time"
+)
+
+type EmailVerifier struct {
+	adminEmailAddress string
+
+	emailSenderService sender.Service
+	codeGenerator      code.Generator
+
+	now func() time.Time
+}
+
+func NewEmailVerifier(
+	adminEmailAddress string,
+	emailSenderService sender.Service,
+	codeGenerator code.Generator,
+	now func() time.Time,
+) *EmailVerifier {
+	verifier := new(EmailVerifier)
+
+	verifier.adminEmailAddress = adminEmailAddress
+	verifier.emailSenderService = emailSenderService
+	verifier.codeGenerator = codeGenerator
+	verifier.now = now
+
+	return verifier
+}
+
+func (v *EmailVerifier) GenerateVerificationCode(user *models.User) string {
+	return v.codeGenerator.GenerateCode(user.Email)
+}
+
+func (v *EmailVerifier) SendVerificationEmail(user *models.User, verificationCode string) error {
```

Comment:
please pass context from caller function

Reply:
What exactly to you mean by context from the caller function in this case? We call this function from service where we retrieved User from the database based on the id and generated the verification code.

Reply:
@ternakkode I too am a bit confused.

I want to go deeper into this conversation because it will force a lot of learning about contexts. After we get your new development environment merged, I'll ask you to explain to Veronika and I what you mean.

I know contexts can be very powerful in building highly parallelizable systems so I want to learn more. I think you'll need to give us an example though of what you mean and so let's get the new development environment up and running and then turn to focus on this PR.

Reply:
sure lets just skip this for now